### PR TITLE
feat(services): migrate ports react query

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-920

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Take rest sometimes

File: libs/pages/application/src/lib/feature/page-settings-ports-feature/crud-modal-feature/crud-modal-feature.tsx [hunk 1]
```diff
@@ -148,50 +145,34 @@ export function CrudModalFeature({ application, onClose, port, organizationId, p
...
-  }
 
-  const onSubmit = methods.handleSubmit((data) => {
-    if (!application) return
+  const onSubmit = methods.handleSubmit(async (data) => {
+    if (!service) return
 
-    setLoading(true)
-    const cloneApplication = handleSubmit(data, application, port)
+    const cloneApplication = handleSubmit(data, service, port)
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/feature/page-settings-ports-feature/crud-modal-feature/crud-modal-feature.tsx:152]: I think you can use generics to properly type request and remove `as` cast.
`handleSubmit` signature should look something like that:
```ts
function handleSubmit<T extends Application | Container, R = T extends Application ? ApplicationEditRequest : ContainerRequest>({
    internal_port,
    external_port,
    publicly_accessible,
    protocol,
    name,
  }: {
    internal_port: string | undefined
    external_port: string | undefined
    publicly_accessible: boolean
    protocol: PortProtocolEnum | undefined
    name: string | undefined
  }, application: T, currentPort?: ServicePort): R {
```

File: libs/pages/application/src/lib/feature/page-settings-ports-feature/crud-modal-feature/crud-modal-feature.tsx [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
+import { ProbeTypeEnum } from '@qovery/shared/enums'
 import { useModal } from '@qovery/shared/ui'
-import { type AppDispatch } from '@qovery/state/store'
 
 export interface CrudModalFeatureProps {
   onClose: () => void
-  organizationId: string
   projectId: string
-  application?: ApplicationEntity
+  service?: Extract<AnyService, Application | Container>
```
Comment (ctjhoa) [inline libs/pages/application/src/lib/feature/page-settings-ports-feature/crud-modal-feature/crud-modal-feature.tsx:23]: Can you prevent `undefined` upfront?

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/application/src/lib/feature/page-settings-ports-feature/crud-modal-feature/crud-modal-feature.tsx (hunk 1 above)
Comment (RemiBonnet) [reply libs/pages/application/src/lib/feature/page-settings-ports-feature/crud-modal-feature/crud-modal-feature.tsx:152]: Yes for simple operation I know I can do that but here we need to update probes for Healthchecks, and I don't want to update all the logic in this PR just only React Query that why I keep the `handleSubmit`

Comment (ctjhoa) [reply libs/pages/application/src/lib/feature/page-settings-ports-feature/crud-modal-feature/crud-modal-feature.tsx]: https://github.com/Qovery/console/pull/1085/commits/81738b0fd981cd8b229870fa3df15f3816998a8f

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**